l10n_br_fiscal/models/document_workflow.py

```diff
@@ -0,0 +1,381 @@
+                        and record.company_state_id.ibge_code
+                        or ""
+                    ),
+                    forma_emissao=1,  # TODO: Implementar campo no Odoo
+                    modelo_documento=record.document_type_id.code or "",
+                    numero_documento=record.document_number or "",
+                    numero_serie=record.document_serie or "",
+                    validar=False,
+                )
+                # TODO: Implementar campos no Odoo
+                # record.key_number = chave_edoc.campos
+                # record.key_formated = ' '.joint(chave_edoc.partes())
+                record.document_key = chave_edoc.chave
+
+    def _document_number(self):
+        self.ensure_one()
+        if self.issuer == DOCUMENT_ISSUER_COMPANY:
+            if self.document_serie_id:
+                self.document_serie = self.document_serie_id.code
+
+                if self.document_type == MODELO_FISCAL_NFSE and not self.rps_number:
+                    self.rps_number = self.document_serie_id.next_seq_number()
+
+                if (
+                    self.document_type != MODELO_FISCAL_NFSE
+                    and not self.document_number
+                ):
+                    self.document_number = self.document_serie_id.next_seq_number()
+
+            if not self.operation_name:
+                self.operation_name = ", ".join(
+                    [line.name for line in self.line_ids.mapped("fiscal_operation_id")]
+                )
+
+            if self.document_electronic and not self.document_key:
+                self._generate_key()
+
+    def _document_confirm(self):
+        self._change_state(SITUACAO_EDOC_A_ENVIAR)
+
+    def action_document_confirm(self):
+        to_confirm = self.filtered(lambda inv: inv.state_edoc != SITUACAO_EDOC_A_ENVIAR)
+        if to_confirm:
+            to_confirm._document_confirm()
+
+    def _no_eletronic_document_send(self):
+        self._change_state(SITUACAO_EDOC_AUTORIZADA)
+
+    def _document_export(self):
+        pass
+
+    def action_document_send(self):
+        to_send = self.filtered(
+            lambda d: d.state_edoc
+            in (
+                SITUACAO_EDOC_A_ENVIAR,
+                SITUACAO_EDOC_ENVIADA,
+                SITUACAO_EDOC_REJEITADA,
+            )
+        )
+        if to_send:
+            to_send._document_send()
+
+    def action_document_back2draft(self):
+        self._change_state(SITUACAO_EDOC_EM_DIGITACAO)
+
+    def _document_cancel(self, justificative):
+        self.ensure_one()
+        self.cancel_reason = justificative
+        if self._change_state(SITUACAO_EDOC_CANCELADA):
+            self.cancel_reason = justificative
+
+    def action_document_cancel(self):
+        self.ensure_one()
+        if self.state_edoc == SITUACAO_EDOC_AUTORIZADA:
+            result = self.env["ir.actions.act_window"]._for_xml_id(
+                "10n_br_fiscal.document_cancel_wizard_action"
```

Comment:
desculpa @rvalyi, mas eu editei errado esse aqui, comi uma letra rss.
```suggestion
                "l10n_br_fiscal.document_cancel_wizard_action"
```

Reply:
Ja faço um amend

Reply:
Beleza, acho que por hora tá tudo ok não tem muito mais o que testar sem o módulo de nfe

Reply:
@netosjb fiz o amend come a correçao